math/vc/include/algebra/math/impl/vc_transform3.hpp

```diff
@@ -0,0 +1,409 @@
+/** Algebra plugins library, part of the ACTS project
+ *
+ * (c) 2020-2021 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Project include(s).
+#include "algebra/common/algebra_qualifiers.hpp"
+#include "algebra/math/impl/vc_vector.hpp"
+
+// Vc include(s).
+#include <Vc/Vc>
+
+// System include(s).
+#include <cassert>
+
+namespace algebra::vc::math {
+
+namespace internal {
+
+/// 4x4 matrix type used by @c algebra::vc::math::transform3
+template <typename scalar_t>
+struct matrix44 {
+
+  /// Equality operator between two matrices
+  bool operator==(const matrix44 &rhs) const {
+    return ((x[0] == rhs.x[0]) && (x[1] == rhs.x[1]) && (x[2] == rhs.x[2]) &&
```

Comment:
This is not using explicit vectorization, when it could actually use the vc ```operator==``` and then a mask check ```isFull()```

Reply:
I believe I borrowed this implementations from one of the other `transform3` classes as well. Since for the `std::array` case this equality was not explicitly declared. And I wonder if the default implementation of `==` for `std::array` is what we actually want here. (I'm honestly not sure.)

I'm 99% sure that the compiler would quite well auto-vectorise this expression, but you're absolutely right. I'll change this as well.

Reply:
if I remember correctly, there was an implementation in ```simd_types.hpp``` somewhere

Reply:
For the Vc matrix type there is.

https://github.com/acts-project/algebra-plugins/blob/main/common/include/algebra/common/simd/simd_types.hpp#L102-L117

I think the equality being evaluated correctly was only ever a concern of mine for `std::array`. As all the other storage types should come with equality operators that behave like we would expect. And `std::array<std::array<FOO, 4>, 4>` may also do this. I just didn't trust that it would... (And not resort to pointer comparison for the outer-array.)

Reply:
Yes, thats the one I meant. It uses the Vc mask types for comparisons. Sorry for hiding it there :).

Ah, OK. I must admit, I don't know what ```std::equal``` would do.

Reply:
Done.